File: app/src/main/java/com/wuyiming/singlecolumnadfeed_android/data/repository/FeedRepository.java
package com.wuyiming.singlecolumnadfeed_android.data.repository;

import com.wuyiming.singlecolumnadfeed_android.data.model.FeedCategory;
import com.wuyiming.singlecolumnadfeed_android.data.model.FeedItem;

import java.util.List;

public interface FeedRepository {
    List<FeedItem> getAllItems();
    List<FeedItem> getItemsByCategory(FeedCategory category);
    List<FeedItem> loadMore(int page, int pageSize, FeedCategory category);
    void reloadData();
    void toggleLike(String feedId);
    void toggleCollect(String feedId);
    FeedItem getItemById(String feedId);
}

File: app/src/main/java/com/wuyiming/singlecolumnadfeed_android/data/repository/FeedRepository.kt
package com.wuyiming.singlecolumnadfeed_android.data.repository

import com.wuyiming.singlecolumnadfeed_android.data.model.AdInsight
import com.wuyiming.singlecolumnadfeed_android.data.model.FeedItem

interface FeedRepository {
    suspend fun loadFeedItems(page: Int, pageSize: Int, refreshSeed: Int): List<FeedItem>
    suspend fun generateAiInsight(item: FeedItem): AdInsight?
    fun toggleLike(item: FeedItem): FeedItem
    fun toggleCollect(item: FeedItem): FeedItem
    fun getItemById(feedId: String): FeedItem?
}
